### PR TITLE
Add test

### LINKED ISSUE #5: Tests
We have no real tests other than to make sure than the `make test` command in the `Makefile`

### PR DESCRIPTION
**Pull request by ykrods**
### Description

Fixes #5

This PR is to add basic test configuration and some examples.

* Based on sphinx test code.
  * [intersphinx example](https://github.com/sphinx-doc/sphinx/blob/v3.4.0/tests/test_ext_intersphinx.py#L231)

* ``make test`` is moved to "docs" env
* ~~Testing on sphinx2 is droped because Sphinx3 has incompatible changes related to testing~~

  * ~~Specifically, [sphinx.testing.path](https://github.com/sphinx-doc/sphinx/blob/v3.4.0/sphinx/testing/path.py) is. it is used by fixtures provided by Sphinx.~~

### CODE REVIEW

**Comment by nabobalis** [comment]
Thanks for this @ykrods, I'll review it over the next 48 hours.

**Comment by nabobalis** [comment]
> * Testing on sphinx2 is droped because Sphinx3 has incompatible changes related to testing

Is there no way to work around this? I would like to still test on 2 as I don't plan to drop support for it anytime.

**Comment by ykrods** [comment]
I found out that it was a simple API change, so I re-added support for sphinx2.

File: tox.ini [hunk 1]
```diff
@@ -1,43 +1,41 @@
...
 
 [testenv]
+deps =
+    sphinx2: sphinx>=2.0,<3.0
+    sphinx3: sphinx>=3.0,<4.0
+    sphinxdev: git+https://github.com/sphinx-doc/sphinx
+extras =
+    test
+commands =
+    py.test
```
Comment (nabobalis) [inline tox.ini:16]:
```suggestion
    pytest
```

File: tox.ini [hunk 2]
```diff
@@ -1,43 +1,41 @@
...
+setenv =
+    PYTHONDONTWRITEBYTECODE = 1
+passenv =
+    HTTP_PROXY
+    HTTPS_PROXY
+    NO_PROXY
+    CIRCLECI
+
+[testenv:docs]
 # We use bash in some of our enviroments so we have to whitelist it.
```
Comment (nabobalis) [inline tox.ini:26]:
```suggestion
```

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks @ykrods!

**Pull request merged**